yt/utilities/answer_testing/framework.py

```diff
@@ -63,6 +81,332 @@
+        # If needed build an instance of the dataset type
+        obj = utils.create_obj(ds, obj_type)
+        determined_field = obj._determine_fields(field)[0]
+        # Get the proper weight field depending on if we're looking at
+        # particles or not
+        if particle_type:
+            weight_field = (determined_field[0], "particle_ones")
+        else:
+            weight_field = ("index", "ones")
+        # Get the average, min, and max
+        avg = obj.quantities.weighted_average_quantity(
+            determined_field,
+            weight=weight_field)
+        minimum, maximum = obj.quantities.extrema(field)
+        # Return as a hashable bytestring
+        return np.array([avg, minimum, maximum])
+
+    def pixelized_projection_values_test(self, ds, axis, field,
+        weight_field=None, dobj_type=None):
+        if dobj_type is not None:
+            obj = utils.create_obj(ds, dobj_type)
+        else:
+            obj = None
+        proj = ds.proj(field, axis, weight_field=weight_field, data_source=obj)
+        frb = proj.to_frb((1.0, 'unitary'), 256)
+        frb[field]
+        if weight_field is not None:
+            frb[weight_field]
+        d = frb.data
+        for f in proj.field_data:
+            # Sometimes f will be a tuple.
+            d["%s_sum" % (f,)] = proj.field_data[f].sum(dtype="float64")
+        # This is to try and remove python-specific anchors in the yaml
+        # answer file. Also, using __repr__() results in weird strings
+        # of strings that make comparison fail even though the data is
+        # the same
+        result = None 
+        for k, v in d.items():
+            k = k.__repr__().encode('utf8')
+            if result is None:
+                result = hashlib.md5(k + v.tobytes())
+            else:
+                result.update(k + v.tobytes())
+        return result.hexdigest()
+
+    def color_conservation_test(self, ds):
+        species_names = ds.field_info.species_names
+        dd = ds.all_data()
+        dens_yt = dd["density"].copy()
+        # Enumerate our species here
+        for s in sorted(species_names):
+            if s == "El": continue
+            dens_yt -= dd["%s_density" % s]
+        dens_yt -= dd["metal_density"]
+        delta_yt = np.abs(dens_yt / dd["density"])
+        # Now we compare color conservation to Enzo's color conservation
+        dd = ds.all_data()
+        dens_enzo = dd["Density"].copy()
+        for f in sorted(ds.field_list):
+            ff = f[1]
+            if not ff.endswith("_Density"):
+                continue
+            start_strings = ["Electron_", "SFR_", "Forming_Stellar_",
+                             "Dark_Matter", "Star_Particle_"]
+            if any([ff.startswith(ss) for ss in start_strings]):
+                continue
+            dens_enzo -= dd[f]
+        delta_enzo = np.abs(dens_enzo / dd["Density"])
+        np.testing.assert_almost_equal(delta_yt, delta_enzo)
```

Comment:
This should probably be moved to the Enzo frontend since it's not saving any answers and isn't really answer-testing specific.